### PR TITLE
Feature/smart health card vaccine info

### PR DESCRIPTION
**Pull request by NC-204**
Move url and base repository to Core module, to be able to use logic in Core.
* Split Api endpoints values and url resolving. Since url resolving cant be done statically in core (since we dont have IoCContainer there).

Add api endpoint for fetching vaccine info for smart health card immunizations.
* Create new coding service and smart health card repository in core to do the fetching
* Add new models for response and request
* Update SMH models for immunizations with equals, id and hashcode. To be able to compare different vaccines.
* Update models tests to all be under the same namespace.

### CODE REVIEW

File: FHICORC.Core/Services/Model/SmartHealthCardModel/Shc/SmartHealthCardImmunization.cs [hunk 1]
```diff
@@ -20,12 +20,6 @@ public class SmartHealthCardImmunization
         public DateTime? OccurrenceDateTime { get; set; }
     }
 
-    public class CodeableConcept
```
Comment (NC-395) [inline FHICORC.Core/Services/Model/SmartHealthCardModel/Shc/SmartHealthCardImmunization.cs:23]: Didn't get it, why did you delete this one?

Comment (NC-204) [reply FHICORC.Core/Services/Model/SmartHealthCardModel/Shc/SmartHealthCardImmunization.cs:23]: Extended the class a bit and it was used by both Immunization and Observation. 
So I moved it into its own file 🙂

Comment (NC-395) [reply FHICORC.Core/Services/Model/SmartHealthCardModel/Shc/SmartHealthCardImmunization.cs:23]: Sure, thanks!
